src/hooks/useLike.ts

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { deleteStoryLike } from '../apis/story';
+import { ERROR_MESSAGES } from '../constants/errorMessages';
+import { ROUTES } from '../constants/routes';
+import { Like } from '../interfaces/like';
+import { postStoryLike } from './../apis/story';
+
+const useLike = (userId: string, storyLikes: Like[], storyId: string) => {
+  const [isLike, setIsLike] = useState(false);
+  const [likeId, setLikeId] = useState('');
+  const [likeCount, setLikeCount] = useState(storyLikes.length);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    for (const storyLike of storyLikes) {
+      if (storyLike.user === userId) {
+        setIsLike(true);
+        setLikeId(storyLike._id);
+        return;
+      }
+    }
```

Comment:
```suggestion
    const findStoryLike = storyLikes.includes((data) => data.user === userId);
    if (findStoryLike) {
      setIsLike(true);
      setLikeId(storyLike._id);
    }
```

혹시 고차함수로 하는건 어떨까? 예를들어서 이건 어때?

Reply:
@tooooo1 오오!! 더 간결하고 가독성도 좋네요 👍 바로 반영할게요!

Reply:
엇.. 24번째줄 아마 문제가 있을거같아요 ㅠㅠ 다시 수정할게요.. 왜냐하면 수정된 곳에는 `storyLike` 변수가 없거든요 ㅠㅠ

```jsx
    const findStoryLike = storyLikes.find((data) => data.user === userId);
    if (findStoryLike) {
      setIsLike(true);
      setLikeId(findStoryLike);
    }
```

이렇게하면 아마 동작할 것 같아요! 확인해 주신 후 반영 부탁드려요!

Reply:
> 엇.. 24번째줄 아마 문제가 있을거같아요 ㅠㅠ 다시 수정할게요.. 왜냐하면 수정된 곳에는 `storyLike` 변수가 없거든요 ㅠㅠ
> 
> ```js
>     const findStoryLike = storyLikes.find((data) => data.user === userId);
>     if (findStoryLike) {
>       setIsLike(true);
>       setLikeId(findStoryLike);
>     }
> ```
> 
> 이렇게하면 아마 동작할 것 같아요! 확인해 주신 후 반영 부탁드려요!

Reply:
@tooooo1 오 로컬에서 시도해보고 제안해주신 대로 반영했어요! 버그 없이 잘 동작합니다 :)